blockstore/splitstore/splitstore.go

```diff
@@ -191,27 +197,49 @@ func (s *SplitStore) DeleteMany(_ []cid.Cid) error {
 	return errors.New("DeleteMany not implemented on SplitStore; don't do this Luke!") //nolint
 }
 
-func (s *SplitStore) Has(cid cid.Cid) (bool, error) {
-	has, err := s.hot.Has(cid)
+func (s *SplitStore) Has(c cid.Cid) (bool, error) {
+	s.txnLk.RLock()
+	defer s.txnLk.RUnlock()
+
+	has, err := s.hot.Has(c)
 
-	if err != nil || has {
+	if err != nil {
 		return has, err
 	}
 
-	return s.cold.Has(cid)
+	if has {
+		err = s.trackTxnRef(c)
+		return true, err
+	}
+
+	return s.cold.Has(c)
 }
 
 func (s *SplitStore) Get(cid cid.Cid) (blocks.Block, error) {
+	s.txnLk.RLock()
+	defer s.txnLk.RUnlock()
+
 	blk, err := s.hot.Get(cid)
 
 	switch err {
 	case nil:
-		return blk, nil
+		err = s.trackTxnRef(cid)
```

Comment:
Are we sure we want to propagate errors from the tx tracking, even if the block was accessed correctly?

Reply:
I am not sure either, we will probably be fine with just warning.

Reply:
done, turned them to warnings and stopped leaking them through the API.